internal/core/src/storage/Event.cpp

```diff
@@ -223,103 +223,120 @@ BaseEventData::BaseEventData(BinlogReaderPtr reader,
     AssertInfo(res.first.ok(), "read payload failed");
     payload_reader = std::make_shared<PayloadReader>(
         res.second.get(), payload_length, data_type, nullable, is_field_data);
-    if (is_field_data) {
-        field_data = payload_reader->get_field_data();
-    }
 }
 
 std::vector<uint8_t>
 BaseEventData::Serialize() {
-    auto data_type = field_data->get_data_type();
-    std::shared_ptr<PayloadWriter> payload_writer;
-    if (IsVectorDataType(data_type) &&
-        !IsSparseFloatVectorDataType(data_type)) {
-        payload_writer = std::make_unique<PayloadWriter>(
-            data_type, field_data->get_dim(), field_data->IsNullable());
+    if (payload_reader->has_binary_payload()) {
+        // for index slice, directly copy payload slice
+        auto payload_size = payload_reader->get_payload_size();
+        auto payload_data = payload_reader->get_payload_data();
+        auto len =
+            sizeof(start_timestamp) + sizeof(end_timestamp) + payload_size;
+        std::vector<uint8_t> res(len);
+        int offset = 0;
+        memcpy(res.data() + offset, &start_timestamp, sizeof(start_timestamp));
+        offset += sizeof(start_timestamp);
+        memcpy(res.data() + offset, &end_timestamp, sizeof(end_timestamp));
+        offset += sizeof(end_timestamp);
+        memcpy(res.data() + offset, payload_data, payload_size);
+        return res;
     } else {
-        payload_writer = std::make_unique<PayloadWriter>(
-            data_type, field_data->IsNullable());
-    }
-    switch (data_type) {
-        case DataType::VARCHAR:
-        case DataType::STRING:
-        case DataType::TEXT: {
-            for (size_t offset = 0; offset < field_data->get_num_rows();
-                 ++offset) {
-                auto str = static_cast<const std::string*>(
-                    field_data->RawValue(offset));
-                auto size = field_data->is_valid(offset) ? str->size() : -1;
-                payload_writer->add_one_string_payload(str->c_str(), size);
-            }
-            break;
```

Comment:
still have to use if-else here, because insert data still uses fieldData to store payload which has not been replaced by bytesBuf, it's reasonable to leave this problem to storageV2 rather than solving this problem in this pr